internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1484,6 +1484,9 @@ func newImageServiceStatefulSet(ctx context.Context, log logrus.FieldLogger, asc
 		var replicas int32 = 1
 		statefulSet.Spec.Replicas = &replicas
 		statefulSet.Spec.Template.Spec.ServiceAccountName = imageServiceName
+		statefulSet.Spec.Template.Spec.SecurityContext = &corev1.PodSecurityContext{
+			RunAsGroup: swag.Int64(0),
```

Comment:
WDYT about parametrizing the whole `SecurityContext` section? We could just offload this responsibility to the user (maybe with some decent default values).

Reply:
Without a more compelling use case (than longhorn in Sylva) I'd prefer to keep this out of our API.

Alternatively, it might also make sense to add API for this if we didn't have a sensible option to hardcode that also solved the problem, but here we are just explicitly stating for outside observers the existing default we were already expecting.